pubtools/_quay/command_executor.py

```diff
@@ -245,3 +258,163 @@ def _run_cmd(self, cmd, err_msg=None, tolerate_err=False, stdin=None):
                 raise RuntimeError(err_msg)
 
         return out_text, err_text
+
+
+class ContainerExecutor(Executor):
+    """Run commands in a Docker container."""
+
+    def __init__(
+        self,
+        image,
+        base_url="unix://var/run/docker.sock",
+        timeout=None,
+        verify_tls=False,
+        cert_path=None,
+    ):
+        """
+        Initialize.
+
+        Args:
+            image (str):
+                Path to an image which will be used for performing skopeo operations. Must be
+                downloadable by Docker.
+            base_url (str):
+                Base URL of the Docker client.
+            timeout (int):
+                Default timeout for API calls, in seconds.
+            verify_tls (bool):
+                Whether to use TLS verification.
+            cert_path (str|None):
+                Custom path to TLS cettificates.
+        """
+        self.image = image
+
+        kwargs = {}
+        kwargs["base_url"] = base_url
+        if timeout:
+            kwargs["timeout"] = timeout
+        if verify_tls:
+            kwargs["base_url"] = kwargs["base_url"].replace("tcp://", "https://")
+            cert_path = cert_path or os.path.join(os.path.expanduser("~"), ".docker")
+            if os.path.isdir(cert_path):
+                kwargs["tls"] = docker.tls.TLSConfig(
+                    client_cert=(
+                        os.path.join(cert_path, "cert.pem"),
+                        os.path.join(cert_path, "key.pem"),
+                    ),
+                    verify=os.path.join(cert_path, "ca.pem"),
+                )
+
+        self.client = docker.APIClient(**kwargs)
+        self.container = self.client.create_container(self.image, detach=True, tty=True)
```

Comment:
What kind of values are you intending to provide as the image ?

If you want the possibility of images being updated after first use, should there be some 'pull' operation here?  On the other hand, the convention could be that you just have to update image in target settings if you ever want it to be updated.  But I have some doubts whether anyone would regularly do that.

Reply:
I added a pull operation. Testing indicates that it's not a very expensive operation if the image is already downloaded.